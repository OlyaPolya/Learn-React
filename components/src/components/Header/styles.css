.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.header-title {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1.5rem 0;
  font-family: 'Satisfy', cursive;
  font-size: 2rem;
  font-weight: bolder;
  background-size: cover ;
  background-position: top;
  animation: changecolor 20s alternate linear infinite;
}

.header-links {
  display: flex;
  justify-content: center;
  margin: 2rem 0;
  gap: 2rem;
  font-size: 1.5rem;
}

.link {
  transition: 0.5s;
}

/* TODO: check if component css variables overwrite App vars 
*/
@keyframes changecolor {
  0% {
    background-color: #f0e4f7;
  }
  20% {
    background-color: #CDB4DB;
  }
  40% {
    background-color: #FFC8DD;
  }
  60% {
    background-color: #FFAFCC;
  }
  80% {
    background-color: #BDE0FE;
  }
  100% {
    background-color: #A2D2FF;
  }
}
