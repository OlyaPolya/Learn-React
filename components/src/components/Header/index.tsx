import React from 'react';
import './styles.css';
import { Link } from '../../types';
import Navigation from './components/Navigation';

type Props = {
  links: Link[];
};

const Header: React.FC<Props> = (props) => (
  <header className="header">
    <p className="header-title">Learning React Project</p>
    <Navigation links={props.links} />
  </header>
);

export default Header;
