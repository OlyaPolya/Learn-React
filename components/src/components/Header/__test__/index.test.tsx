import React from 'react';
import { render } from '@testing-library/react';
import Header from '..';
import { MemoryRouter } from 'react-router-dom';
import fixtures from './fixtures';

describe('<Header/>', () => {
  it('should match snapshot', () => {
    const component = render(
      <MemoryRouter>
        <Header {...fixtures.props} />
      </MemoryRouter>
    );

    expect(component.container).toMatchSnapshot();
  });
});
