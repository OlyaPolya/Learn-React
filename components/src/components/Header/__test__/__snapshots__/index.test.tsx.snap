// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <p
      class="header-title"
    >
      Learning React Project
    </p>
    <ul
      class="header-links"
    >
      <li
        class="link"
      >
        <a
          href="/"
        >
          Test1
        </a>
      </li>
      <li
        class="link"
      >
        <a
          href="/test"
        >
          Test2
        </a>
      </li>
    </ul>
  </header>
</div>
`;
